### ISSUE TITLE
need help with defining Elit Tracker, problem with login.php and 403 forbidden

### ISSUE THREAD

**Issue by garfield69** [Jackett/Jackett#2042]
I'm creating a definition for Elit Tracker @ http://elittorent.us and I have run into a problem with the login.
The log shows that I am receiving a 403 Forbidden when I attempt to access http://elittorent.us/login.php

This is the definition so far:
```yaml
---
  site: elittracker
  name: Elit Tracker
  description: "Elit Tracker (ET) is a HUNGARIAN Private Torrent Tracker for 0DAY / GENERAL"
  language: hu-hu
  type: private
  encoding: iso-8859-2
  links:
    - http://elittorent.us/

  caps:
    categorymappings:
      - {id: 35, cat: Movies, desc: "BluRay-Egyéb"}
      - {id: 30, cat: Movies, desc: "BluRay-Hun"}
      - {id: 34, cat: Movies, desc: "Cam-Egyéb"}
      - {id: 33, cat: Movies, desc: "Cam-Hun"}
      - {id: 36, cat: Movies, desc: "DVD-Egyéb"}
      - {id: 2, cat: Movies, desc: "DVD-Eng"}
      - {id: 3, cat: Movies, desc: "DVD-Hun"}
      - {id: 6, cat: Other, desc: "Egyéb"}
      - {id: 10, cat: Console, desc: "Játék-Konzol"}
      - {id: 7, cat: Console, desc: "Játék-PC"}
      - {id: 8, cat: Console, desc: "Játék-PS"}
      - {id: 9, cat: Console, desc: "Játék-Xbox"}
      - {id: 11, cat: Other, desc: "Klip"}
      - {id: 4, cat: Books, desc: "Könyv-Eng"}
      - {id: 5, cat: Books, desc: "Könyv-Hun"}
      - {id: 12, cat: Books, desc: "Mese-Eng"}
      - {id: 13, cat: Books, desc: "Mese-Hun"}
      - {id: 14, cat: PC, desc: "Mobil"}
      - {id: 15, cat: PC, desc: "Program"}
      - {id: 16, cat: PC, desc: "Program-Linux"}
      - {id: 17, cat: TV, desc: "Sorozat-Eng"}
      - {id: 18, cat: TV, desc: "Sorozat-Hun"}
      - {id: 37, cat: Movies, desc: "XviD-Egyéb"}
      - {id: 19, cat: Movies, desc: "XviD-Eng"}
      - {id: 20, cat: Movies, desc: "XviD-Hun"}
      - {id: 21, cat: XXX, desc: "XXX"}
      - {id: 22, cat: XXX, desc: "XXX-Képek"}
      - {id: 32, cat: Music, desc: "Zene-Egyéb"}
      - {id: 23, cat: Music, desc: "Zene-Eng"}
      - {id: 24, cat: Music, desc: "Zene-Hun"}

    modes:
      search: [q]
      tv-search: [q, season, ep]
      movie-search: [q]

  login:
    path: login.php
    method: post
    inputs:
      username: "{{ .Config.username }}"
      password: "{{ .Config.password }}"
      logout: "yes"
      logintype: "yes"
      
    error:
      - selector: table tr td.thead
        message:
          selector: td:nth-child(2)
    test:
      path: browse.php
      selector: div span a:nth-child(2)[href*="/logout.php?logouthash="]

  search:
    paths:
      - path: browse.php
        method: post
    inputs:
      do: "search"
      keywords: "{{ .Keywords }}"
      search_type: "t_name"
      # multi cat is not supported so defaulting to ALL
      category: "0"
      submit: "Keresés!"

    rows:
      selector: table > tbody > tr:has(td a[href*="/browse.php?browse_categories&amp;category="])
    fields:
      title:
        selector: td:nth-of-type(3) a
        attribute: title
      details:
        selector: tdnth-of-type(2) a:nth-of-type(2)
        attribute: href
      download:
        selector: td:nth-of-type(3) a
        attribute: href
      category:
        selector: td:nth-of-type(1) a
        attribute: href
        filters:
          - name: querystring
            args: category
      date:
        selector: td:nth-of-type(2)
        filters:
          - name: replace
            args: ["Hozzáadva:", ""]
          - name: append
            args: " +01:00"
          - name: dateparse
            args: "2006 02, January, 15:04:05 -07:00"
      files:
        selector: td:nth-of-type(4) a
      size:
        selector: td:nth-of-type(9)
      seeders:
        selector: td:nth-of-type(6)
      leechers:
        selector: td:nth-of-type(7)
      grabs:
        selector: td:nth-of-type(9) a b
      downloadvolumefactor:
        case:
#          img[src="pic/diamond.png"]: "0"
          img[src$="pic/freedownload.gif"]: "0"
          img[src$="pic/silverdownload.gif"]: "0.5"
#          img[src="pic/bronze.gif"]: "0.75"
          "*": "1"
      uploadvolumefactor:
        case:
#          img[src="pic/diamond.png"]: "2"
          "*": "1"
```
here is the login.html I am trying to access:
[elittracker.login.html.txt](https://github.com/Jackett/Jackett/files/1426232/elittracker.login.html.txt)
and here is the log:
[elittracker.log](https://github.com/Jackett/Jackett/files/1426233/elittracker.log)
:sos:

**Comment by halali** [comment]
isnt it 
```
  login:
    path: takelogin.php
    method: post
    inputs:
      username: "{{ .Config.username }}"
      password: "{{ .Config.password }}"
      logout: "yes"
      logintype: "yes"
```

**Comment by garfield69** [comment]
I tried that also, but I got the same 403 forbidden error :disappointed:

**Comment by halali** [comment]
Have same result,

probably they have blocked Jackett

**Comment by garfield69** [comment]
Darn. Thank you for taking a look.

**Comment by maxcore** [comment]
work perfect for me!

**Comment by kaso17** [comment]
You might want to try form login, they're setting a session cookie which might be needed.
```
  login:
    path: login.php
    method: form
    form: form[action="takelogin.php"]
    inputs:
      username: "{{ .Config.username }}"
      password: "{{ .Config.password }}"
```

**Comment by garfield69** [comment]
I suspect you are right about the session cookie....
when using the browser to access http://elittorent.us/ directly, the chrome browser development tools network page first sees this:
![image](https://user-images.githubusercontent.com/6933240/32473688-c1b523fe-c3cd-11e7-8af3-d7f2d92ea7b8.png)
then the login.php is presented:
![image](https://user-images.githubusercontent.com/6933240/32473796-281cdede-c3ce-11e7-9074-2739511d7899.png)

I tried the method: form but still got the 403 forbidden.
Any ideas on how to proceed?

**Comment by kaso17** [comment]
They need the Accept-Language header set, without it they return an internal server error.
fixed via 8c053797ce417d58c2464dbc7cad8cf646a3c2b8

This login block works for me:
```
  login:
    path: login.php
    method: form
    form: form[action="takelogin.php"]
    inputs:
      username: "{{ .Config.username }}"
      password: "{{ .Config.password }}"
    error:
      - selector: td:contains("Hiba")
    test:
      path: browse.php
```

**Comment by garfield69** [comment]
Thank you, that worked a treat :-)
